tests/scripts/thread-cert/network_diag.py

```diff
@@ -0,0 +1,335 @@
+                f'if_in_unknown_protos={self.if_in_unknown_protos}, ' +
+                f'if_in_errors={self.if_in_errors}, ' +
+                f'if_out_errors={self.if_out_errors}, ' +
+                f'if_in_ucast_pkts={self.if_in_ucast_pkts}, ' +
+                f'if_in_broadcast_pkts={self.if_in_broadcast_pkts}, ' +
+                f'if_in_discards={self.if_in_discards}, ' +
+                f'if_out_ucast_pkts={self.if_out_ucast_pkts}, ' +
+                f'if_out_broadcast_pkts={self.if_out_broadcast_pkts}, ' +
+                f'if_out_discards={self.if_out_discards})')
+
+
+class MacCountersFactory:
+
+    def parse(self, data, message_info):
+        return MacCounters(struct.unpack('>9I', data.read(4 * 9)))
+
+
+class BatteryLevel:
+
+    def __init__(self, battery_level: int):
+        self._battery_level = battery_level
+
+    @property
+    def battery_level(self):
+        return self._battery_level
+
+    def __eq__(self, other):
+        common.expect_the_same_class(self, other)
+
+        return self.battery_level == other.battery_level
+
+    def __repr__(self):
+        return f'BatteryLevel(battery_level={self.battery_level})'
+
+
+class BatteryLevelFactory:
+
+    def parse(self, data, message_info):
+        return BatteryLevel(struct.unpack('>B', data.read(1))[0])
+
+
+class SupplyVoltage:
+
+    def __init__(self, supply_voltage: int):
+        self._supply_voltage = supply_voltage
+
+    @property
+    def supply_voltage(self):
+        return self._supply_voltage
+
+    def __eq__(self, other):
+        common.expect_the_same_class(self, other)
+
+        return self.supply_voltage == other.supply_voltage
+
+    def __repr__(self):
+        return f'SupplyVoltage(supply_voltage={self.supply_voltage})'
+
+
+class SupplyVoltageFactory:
+
+    def parse(self, data, message_info):
+        return SupplyVoltage(struct.unpack('>H', data.read(2))[0])
+
+
+class ChildTableEntry:
+
+    def __init__(self, timeout: int, child_id: int, mode: mle.Mode):
+        self._timeout = timeout
+        self._child_id = child_id
+        self._mode = mode
+
+    @property
+    def timeout(self):
+        return self._timeout
+
+    @property
+    def child_id(self):
+        return self._child_id
+
+    @property
+    def mode(self):
+        return self._mode
+
+    def __eq__(self, other):
+        common.expect_the_same_class(self, other)
+
+        return (self.timeout == other.timeout and
+                self.child_id == other.child_id and self.mode == other.mode)
+
+    def __repr__(self):
+        return f'ChildTableEntry(timeout={self.timeout}, child_id={self.child_id}, mode={self.mode})'
+
+
+class ChildTable:
+
+    def __init__(self, children: [ChildTableEntry]):
+        self._children = children
+
+    @property
+    def children(self):
+        return self._children
+
+    def __eq__(self, other):
+        common.expect_the_same_class(self, other)
+
+        return self.children == other.children
```

Comment:
Should same children info but different order be considered equal?  It is not currently.

Reply:
Resolved by sorting children by ID.